src/compiler/crystal/semantic/semantic_visitor.cr

```diff
@@ -72,6 +72,10 @@ abstract class Crystal::SemanticVisitor < Crystal::Visitor
     message = "can't find file '#{ex.filename}'"
     notes = [] of String
 
+    ex.similar_require.tap do |similar|
```

Comment:
I wasn't sure what the preference would be for compiler code...
Should I use something like this, or would it be better to simply memoize `NotFoundError#similar_require`